docs/schulcloud-server/Coding-Guidelines/domain-object-validation.md

```diff
@@ -0,0 +1,20 @@
+# Domain Object Validation
+
+If you need to validate a domain object, please write an independent class, so that the domain object itself, its repo and services can reuse it.
+
+Eric Evans suggests using the specification pattern.
+A specification fulfills the following interface:
+
+```typescript
+public interface Specification<T> {
+    boolean isSatisfiedBy(T t);
+}
+```
+
+A specification checks if a domain object fulfills the conditions of the specification.
+
+A specification can simply specify that a domain object is valid. E.g. a `Task` has an owner and a description.
```

Comment:
Any hind how we use it? 
Use methods that do more like simple getter/setter (to make class cohesion vsisible), props a private. Only methods that can handle by the domain object. It is allow that other domain object can pass by in props, but should not generate a new one.